src/main/python/hydra/lib/hydrabase.py

```diff
@@ -111,8 +119,48 @@ def delete_tasks_from_group(self, num_tasks_to_delete):
                 l.info("Error in deletiong task: %s " % str(e))
                 return 1
         self.hydra.refresh_app_info(self.app_name)
+
+        # Make sure that desired number of tasks have been deleted.
+        num_tasks_after_delete = self.get_num_tasks()
+        if current_num_tasks_in_grp - num_tasks_to_delete != num_tasks_after_delete:
+            l.error("Error in deleting tasks from group")
+            return 1
         return deleted_tasks
 
+    def wait_for_plan_done(self, timeout=30, sleep_before_next_try=0.2):
```

Comment:
So, each group has multiple tasks. Each task has a plan. This function is verifying that all tasks in group have finished their work.
Do you want me to write this function on task level (since individual task has plan)?